docs/src/intro/usage.md

```diff
@@ -0,0 +1,54 @@
+# Usage
+Before you can use Miden VM, you'll need to make sure you have Rust [installed](https://www.rust-lang.org/tools/install). Miden VM v0.2, requires Rust version **1.62** or greater.
+
+Miden VM consists of several crates, each of which exposes a small set of functionality. The most notable of these crates are:
+* [miden-processor](https://crates.io/crates/miden-processor), which can be used to execute Miden VM programs.
+* [miden-prover](https://crates.io/crates/miden-prover), which can be used to execute Miden VM programs and generate proofs of their execution.
+* [miden-verifier](https://crates.io/crates/miden-verifier), which can be used to verify proofs of program execution generated by Miden VM prover.
+
+The above functionality is also exposed via the single [miden](https://crates.io/crates/miden) crate, which also provides a CLI interface for interacting with Miden VM.
+
+## CLI interface
+
+### Compiling Miden VM
+To compile Miden VM to an executable file, you can run the following command:
+```
+cargo build --release --features executable
+```
+This will place `miden` executable in the `./target/release` directory.
+
+By default, the executable will be compiled in the single-threaded mode. If you would like to enable multi-threaded proof generation, you can compile Miden VM using the following command:
+```
+cargo build --release --features "executable concurrent"
+```
+
+### Controlling parallelism
+Internally, Miden VM uses [rayon](https://github.com/rayon-rs/rayon) for parallel computations. To control the number of threads used to generate a STARK proof, you can use `RAYON_NUM_THREADS` environment variable.
+
+### Running Miden VM
+Once the executable has been compiled, you can run Miden VM like so:
+```
+./target/release/miden [subcommand] [parameters]
+```
+Currently, Miden VM can be executed with the following subcommands:
+* `run` - this will execute a Miden assembly program and output the result, but will not generate a proof of execution.
+* `prove` - this will execute a Miden assembly, and will also generate a STARK proof of execution.
+* `verify` - this will verify a previously generated proof of execution for a given program.
+* `compile` - this will compile a Miden assembly program (i.e., build a program [MAST](../design/programs.md)) and outputs stats about the compilation process.
+* `analyze` - this will run a Miden assembly program against specific inputs and will output stats about its execution.
+
+All of the above subcommands require various parameters to be provided. To get more detailed help on what is needed for a given subcommand, you can run the following:
+```
+./target/release/miden [subcommand] --help
+```
+For example:
+```
+./target/release/miden prove --help
+```
+
```

Comment:
Do you mean compiling in debug mode? If so, yes, it is available, but I'm imagining that there is not much sense of running the VM in the debug mode for anyone who's not working on the VM itself. Plus, running in debug mode will be 40x slower than in release mode.

Reply:
I mean being able to see the VM state at each step via our internal "debug mode" (i.e. spitting out the list of execute_iter's states or similar), but it looks like only analysis output is available. I could imagine it being interesting & useful for someone who is exploring the VM or writing Miden assembly code to see the state of the stack, etc. I'm sure there won't be many, but Miden assembly is currently all we have available. There's no change needed here - I just thought that it would be useful to inform people if it was available (but it could be in the repo README anyway). Also, as discussed, maybe we'll make a separate web tool at some point where people could see this info